src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/attribute_editor.cljs

```diff
@@ -1,121 +1,95 @@
-              (swap! state assoc :editing? false)))))
 
+(react/defc WorkspaceAttributeViewerEditor
+  {:get-attributes
+   (fn [{:keys [state]}]
+     (let [{:keys [attributes]} @state
+           duplicates (not-empty (utils/find-duplicates (map key attributes)))
+           any-empty? (some (fn [[k v]]
+                              (let [[ek ev] (map (comp empty? trim) [k v])]
+                                (or ek ev)))
+                            attributes)
+           with-spaces (->> attributes
+                            (map (comp trim key))
+                            (filter (partial re-find #"\s"))
+                            not-empty)]
+       (cond duplicates {:error (str "Duplicate keys: " (join ", " duplicates))}
+             any-empty? {:error "Empty keys and values are not allowed."}
+             with-spaces {:error (str "Keys cannot have spaces: " (join ", " with-spaces))}
+             :else {:success (into {} attributes)})))
+   :render
+   (fn [{:keys [props state after-update]}]
+     (let [{:keys [editing?]} props]
+       [:div {}
+        (style/create-section-header "Workspace Attributes")
+        (style/create-paragraph
+          [:div {}
+           (when editing?
+             [:div {:style {:marginBottom "0.25em"}}
+              [comps/Button {:icon :add :text "Add new"
+                             :onClick (fn [_]
+                                        (swap! state update :attributes conj ["" ""])
+                                        ;; have to do this by ID not ref, since the fields are generated within Table
+                                        (after-update #(.focus (.getElementById js/document "focus"))))}]])
+           [table/Table
+            {:key editing?
+             :reorderable-columns? false :sortable-columns? (not editing?) :filterable? false :pagination :none
+             :empty-message "No Workspace Attributes defined"
+             :row-style {}
+             :always-sort? (not editing?)
+             :header-row-style {:borderBottom (str "2px solid " (:line-gray style/colors))
+                                :backgroundColor "white" :color "black" :fontWeight "bold"}
+             :resize-tab-color (:line-gray style/colors)
+             :columns (if editing?
+                        [{:starting-width 40 :resizable? false :as-text (constantly "Delete")
+                          :content-renderer
+                          (fn [index]
+                            (icons/icon {:style {:color (:exception-red style/colors)
+                                                 :verticalAlign "middle"
+                                                 :cursor "pointer"}
+                                         :onClick #(swap! state update :attributes utils/delete index)}
+                                        :delete))}
+                         {:header "Key" :starting-width 300 :as-text (constantly nil)
```

Comment:
Do you have to add the `(constantly nil)` because `:as-text` is required? Maybe that should be the default?